velox/functions/lib/window/NthValue.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <string>
+
+namespace facebook::velox::functions::window {
+void registerBigintNthValue(const std::string& name);
```

Comment:
These functions need to be documented as it is not clear what 'bigint' and 'integer' refer to. Looking at the implementation it appears these are the type of the 'offset' parameter. However, there are more differences then just types. Bigint version allows non-constant offsets while integer version doesn't allow it. This is confusing and non-obvious to the reader.

Instead of providing these functions, I suggest that lib contains a base implementation of nth-value function and spark and presto folders contain the final version + registration logic.

Reply:
From this and below comments, looks like I need to do more refactor around this, please let me know I should raise another PR for this.
Thanks.

Reply:
@zhli1142015 Yes, please, refactor in a new PR.